pkg/rolie/clone.go

```diff
@@ -15,26 +16,40 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func Clone(URI string, dir string) error {
+func Clone(URI string, dir string, filter string) error {
 	f := fetcher{
 		URI:           URI,
 		DirectoryPath: dir,
+		Filter:        filter,
+	}
+	err := f.Init()
+	if err != nil {
+		return err
 	}
-	f.Init()
 	return f.Clone()
 }
 
 type fetcher struct {
 	URI           string
 	BaseURI       string
 	DirectoryPath string
+	Filter        string
+	Re            *regexp.Regexp
 }
 
-func (f *fetcher) Init() {
+func (f *fetcher) Init() error {
 	idx := strings.LastIndex(f.URI, "/")
 	if idx != -1 && idx != len(f.URI) {
 		f.BaseURI = f.URI[:idx]
 	}
+	if f.Filter != "" {
+		re, err := regexp.Compile(f.Filter)
+		if err != nil {
+			return fmt.Errorf("could not compile reg-exp %s: %s", f.Filter, err)
```

Comment:
nitpick: `regex` seems cleaner instead of `reg-exp`

Reply:
Fixed thx!